sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1163,13 +1163,13 @@ private String getLinkName(String sessionId) {
     }
 
     /**
-     * Map the error to {@link ServiceBusAmqpException}
+     * Map the error to {@link ServiceBusReceiverException}
      */
     private Throwable mapError(Throwable throwable, ServiceBusErrorSource errorSource) {
-        if ((throwable instanceof ServiceBusAmqpException) || !(throwable instanceof AmqpException)) {
+        if ((throwable instanceof ServiceBusReceiverException) || !(throwable instanceof AmqpException)) {
             return throwable;
         } else {
-            return new ServiceBusAmqpException((AmqpException) throwable, errorSource);
+            return new ServiceBusReceiverException((AmqpException) throwable, errorSource);
         }
```

Comment:
This can be simplified 
```java
if (throwable instanceof AmqpException) { 
   return new ServiceBusReceiverException((AmqpException) throwable, errorSource);
}
return throwable;
```